src/webview/js/js.js

```diff
@@ -1009,3 +1012,16 @@ documentBody.addEventListener('touchmove', (e: TouchEvent) => {
 documentBody.addEventListener('drag', (e: DragEvent) => {
   clearTimeout(longPressTimeout);
 });
+
+// It's possible we could call this earlier, and would see some performance
+// benifit from doing so, since js.js takes about 16ms to run on a Pixel 3a.
+// However, I don't see that as being worth the possible bugs from things
+// loading too early.
+sendMessage({ type: 'ready' });
+
+// Keep retrying sending the ready event, in case the first one is sent while
+// the queue isn't ready. While this isn't something I've observed in testing,
+// we've previously had bugs that were caused by this (for instance, #3078)
+readyRetryInterval = setInterval(() => {
+  sendMessage({ type: 'ready' });
+}, 100);
```

Comment:
I'd be a bit more comfortable seeing these grouped together in a function.

I'd also quite like to have this logic located next to the `clearInterval` and the declaration of `readRetryInterval`, so that all that logic is in one place.

How about sticking these in a function up there, and then invoking that function from here? Could be named like `signalReadyForEvents`.

Reply:
Done :)